src/firefly/js/ui/NaifidPanel.jsx

```diff
@@ -108,80 +100,44 @@ NaifidPanelView.propTypes = {
     popStyle : PropTypes.object, //style for the suggestion box popup list
     onUnmountCB : PropTypes.func,
     feedbackStyle: PropTypes.object,
-    fireValueChange: PropTypes.func
+    fireValueChange: PropTypes.func,
+    updateNaifNameValue: PropTypes.func
 };
 
-function valueChanged(payload,fireValueChange){
-    let showHelp = false;
-    if(payload.value ==='') showHelp = true;
-    fireValueChange({message:payload.message, valid:payload.valid, displayValue:payload.value, showHelp});
-}
-
-function updateFeedback(naifObj, valid, fireValueChange){
-    const payload={
-        feedback: `Object Name: <b>${naifObj.name}</b>, NAIF ID: <b>${naifObj.naifid}</b>`,
-        valid,
-        displayValue: naifObj.name,
-        value: naifObj.name+';'+naifObj.naifid,//this is the returned value from the component.
-    };
-    naifNamevalue = naifObj.name;
-    fireValueChange(payload);
-}
-
-
-/**
- * Parses the selected option, and returns the value which gets populated in the input field of the suggestion box
- * @param onCallBack
- * @returns {function(*, *=)}
- */
-function getNaifidValue(onCallBack) {
-    return (val, obj) => {
-        if (! obj) return;
-        onCallBack(obj);
-        return obj.name;
-    };
-}
-
-
-/**
- * renders suggestion list's popup
- * @param naifObj
- * @returns {*}
- */
-function renderSuggestion(naifObj){
-    return  <span>Name:<b>{naifObj.name}</b>, NAIF ID: <b>{naifObj.naifid}</b></span>;
-}
-
-
-function handleValueChange(payload, fireValueChange) {
-    const newPayload= {...payload};
-    newPayload.valid  = Boolean(newPayload.value) || (newPayload.displayValue === '' || newPayload.displayValue === naifNamevalue);
-    if (!newPayload.valid) {
-        if (!newPayload.message) {
-            newPayload.message = 'Please use name from the list';
-        }
-        if (naifNamevalue) {
-            newPayload.value = '';
-            newPayload.feedback = '';
-        }
-    }
-    fireValueChange(newPayload) ;
-}
 
 export const NaifidPanel= memo( (props) => {
+    const {fieldKey='NaifId'} = props;
+    const {viewProps, fireValueChange} = useFieldGroupConnector({fieldKey, ...props});
+
+    const naifNameValue = useRef(props.value ?? ''); // ref instead of state so that we can mutate it without causing re-renders
```

Comment:
I had a little trouble understanding this. So I think `naifNameValue` was a global before and you are now keeping track of it as a ref. OK - this makes sense. `naifNameValue` should not have been a global.

Reply:
Yes that's right! See [this diff](https://github.com/Caltech-IPAC/firefly/commit/c072d51a2eab2ac1bc759dd4de71a2adeb7f306f)